fly/commands/clear_resource_cache.go

```diff
@@ -0,0 +1,41 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/fly/commands/internal/flaghelpers"
+	"github.com/concourse/concourse/fly/rc"
+)
+
+type ClearResourceCacheCommand struct {
+	Resource 	 flaghelpers.ResourceFlag `short:"r" long:"resource" required:"true" value-name:"PIPELINE/RESOURCE" description:"Name of a resource to clear cache"`
+	Version      *atc.Version             `short:"v" long:"version"                  value-name:"VERSION"           description:"Version of the resource to check from, e.g. digest:sha256@..."`
```

Comment:
Maybe we should mention in the description that if you don't add a Version, it will result in all the resource caches for that resource being deleted. I think this might be something we want to think about, whether we want to allow all the resource caches to be deleted when they don't pass a version. For example, a user can forget to add a `--version` flag when they are running the command and did not intend to delete all the resource caches.

If we are worried about this happening, maybe we can accept a special `all` value for the version flag and only if `all` is passed then we will delete all the caches. If the flag is not specified then it will error.